frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -230,24 +230,34 @@ const GitSection: React.FC<GitSectionProps> = ({
         return;
       }
       const detectedGitType = detectGitType(url);
-      const gitType = values.git.showGitType ? values.git.type : detectedGitType;
       const gitRepoName = formType !== 'sample' && detectGitRepoName(url);
 
       // Updated detectedType only
       if (detectedGitType !== values.git.detectedType) {
-        setFieldValue('git.detectedType', gitType);
+        setFieldValue('git.detectedType', detectedGitType);
       }
       if (detectedGitType === GitTypes.unsure && !values.git.showGitType) {
         setFieldValue('git.showGitType', true);
       }
+
+      if (
+        detectedGitType === GitTypes.github ||
+        detectedGitType === GitTypes.gitlab ||
+        detectedGitType === GitTypes.bitbucket
+      ) {
```

Comment:
Does this also if we compare it to "not unsure" (aka sure) instead of list all (currently) known Git provider?

You can optional add the && showGitTypes so that `setFieldValue` is not called if it's already false.

```suggestion
      } else if (detectedGitType !== GitTypes.unsure && values.git.showGitType) {
```

Reply:
i am now using a cleaner approach to the problem. Updated